rust/datafusion/src/logical_plan/expr.rs

```diff
@@ -158,6 +158,15 @@ pub enum Expr {
         /// List of expressions to feed to the functions as arguments
         args: Vec<Expr>,
     },
+    /// Returns whether the list contains the expr value.
+    InList {
+        /// The value to compare
+        expr: Box<Expr>,
+        /// The low end of the range
+        list: Vec<Expr>,
+        /// Whether the expression is negated
+        negated: bool,
```

Comment:
We might keep negated out and use `not` instead?

Reply:
This helps keeping the logical plan simple, and also makes future code that uses the LP tree simple, e.g. an optimization rule on `not(..)`

Reply:
I mainly included `negated` to allow pretty printing like: `'z' NOT IN ('x','y')`. I have changed this so it now uses the `not` `expr` so will now display `NOT 'z' IN ('x','y')`

Reply:
I think supporting sql style `NOT IN` would be nice (though no changes needed in this PR)

Reply:
Would be nice indeed for next PR, I think we could have a special case to match on Not (ListIn (...) in the formatter instead :+1:

Reply:
I can't remember exactly, but I think there might be some semantic difference (regarding NULLs, of course) in SQL between `c NOT IN (..)` and `NOT c IN (....)` FWIW that might require representing them differnetly

Reply:
hm ok... in that case my initial suggestion might have been wrong... would good to have some tests for this

Reply:
Thanks for the comments. I have done some testing with Postgres 13.1 and found that it does not appear to make a difference. These are all equivalent and return `NULL`.

```sql
SELECT NOT NULL IN ('a');
SELECT NULL NOT IN ('a');
SELECT NOT 'a' IN (NULL);
SELECT 'a' NOT IN (NULL);
```

Reply:
Ok, thanks @seddonm1 for checking . sounds good to me